hr_timesheet_purchase_order/models/hr_employee_base.py

```diff
@@ -8,3 +12,237 @@ class HrEmployeeBase(models.AbstractModel):
+        ],
+        default="date",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    mon = fields.Boolean(string="Mon", compute="_compute_repeat", readonly=False)
+    tue = fields.Boolean(string="Tue", compute="_compute_repeat", readonly=False)
+    wed = fields.Boolean(string="Wed", compute="_compute_repeat", readonly=False)
+    thu = fields.Boolean(string="Thu", compute="_compute_repeat", readonly=False)
+    fri = fields.Boolean(string="Fri", compute="_compute_repeat", readonly=False)
+    sat = fields.Boolean(string="Sat", compute="_compute_repeat", readonly=False)
+    sun = fields.Boolean(string="Sun", compute="_compute_repeat", readonly=False)
+
+    repeat_day = fields.Selection(
+        [(str(i), str(i)) for i in range(1, 32)],
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_week = fields.Selection(
+        [
+            ("first", "First"),
+            ("second", "Second"),
+            ("third", "Third"),
+            ("last", "Last"),
+        ],
+        default="first",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_weekday = fields.Selection(
+        [
+            ("mon", "Monday"),
+            ("tue", "Tuesday"),
+            ("wed", "Wednesday"),
+            ("thu", "Thursday"),
+            ("fri", "Friday"),
+            ("sat", "Saturday"),
+            ("sun", "Sunday"),
+        ],
+        string="Day Of The Week",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_month = fields.Selection(
+        [
+            ("january", "January"),
+            ("february", "February"),
+            ("march", "March"),
+            ("april", "April"),
+            ("may", "May"),
+            ("june", "June"),
+            ("july", "July"),
+            ("august", "August"),
+            ("september", "September"),
+            ("october", "October"),
+            ("november", "November"),
+            ("december", "December"),
+        ],
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    repeat_show_dow = fields.Boolean(compute="_compute_repeat_visibility")
+    repeat_show_day = fields.Boolean(compute="_compute_repeat_visibility")
+    repeat_show_week = fields.Boolean(compute="_compute_repeat_visibility")
+    repeat_show_month = fields.Boolean(compute="_compute_repeat_visibility")
+
+    @api.depends(
+        "is_auto_po_generate", "repeat_unit", "repeat_on_month", "repeat_on_year"
+    )
+    def _compute_repeat_visibility(self):
+        """Based on the selected parameters sets
+        the fields that should be visible to the user
+        """
+        for item in self:
+            item.repeat_show_day = (
+                item.is_auto_po_generate
+                and (item.repeat_unit == "month" and item.repeat_on_month == "date")
+                or (item.repeat_unit == "year" and item.repeat_on_year == "date")
+            )
+            item.repeat_show_week = (
```

Comment:
This compute value is the same at before, maybe use lambda function or separate for remove code duplication.